app/models/user.rb

```diff
@@ -7,6 +7,51 @@ class User < ApplicationRecord
   validates :name, presence: true, length: { maximum: 20 }
 
   has_many :posts
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships
+  has_many :reverse_friendship, class_name: 'Friendship', foreign_key: 'friend_id'
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :pending_request, -> { where confirm: false }, class_name: 'Friendship', foreign_key: 'friend_id'
```

Comment:
**OPTIONAL**
Complete this with 
```
has_many :pending_friends, through: :pending_request, source: :friend
```
To have users waiting you confirm the friendship